### ISSUE TITLE
[OS X] Can't share by link using password 

### ISSUE THREAD

**Issue by lmaestro** [owncloud/client#2917]
### Actual behaviour
Error 400 
![screen shot 2015-03-04 at 11 16 48 am](https://cloud.githubusercontent.com/assets/4689430/6481723/3473239a-c260-11e4-8cfa-ad231b03babb.png)

### Steps to reproduce
1. Launch the app
2. Select to sync a local folder
3.Click on Share by link
3. Click on Set password -> introduce the password

### Server configuration

{"installed":true,"maintenance":false,"version":"8.0.1.0","versionstring":"8.0.1","edition":"Enterprise"}

Note: Encryption is enable

### Client configuration
Client version: version 1.8.-nightly20150303 (build 2072)

Operating system:OS X 10.10

### Logs
```
03-04 11:15:57:372 0x104d7e940 OCC::FolderMan::slotRunOneEtagJob: Scheduling "ownCloud" to check remote ETag
03-04 11:15:57:373 0x104d7e940 OCC::AbstractNetworkJob::start: !!! OCC::RequestEtagJob created for  QUrl( "https://docker.solidgear.es:50013" )  querying ""
03-04 11:15:57:537 0x104d7e940 OCC::Folder::etagRetreived: * Compare etag with previous etag: last: ""54f6cb7f38bab"" , received: ""54f6cb7f38bab""
03-04 11:15:57:709 0x104d7e940 OCC::ownCloudGui::slotComputeOverallSyncStatus: Folder in overallStatus Message:  OCC::Folder(0x10cf0d1d0)  with name  "ownCloud"
03-04 11:15:57:711 0x104d7e940 OCC::FolderMan::slotRunOneEtagJob: No more remote ETag check jobs to schedule.
03-04 11:16:00:144 0x104d7e940 OCC::AbstractNetworkJob::start: !!! OCC::OcsShareJob created for  QUrl( "https://docker.solidgear.es:50013" )  querying ""
03-04 11:16:00:237 0x104d7e940 OCC::ShareDialog::slotPasswordSet: void OCC::ShareDialog::slotPasswordSet(const QString &) Status code:  400
03-04 11:16:00:237 0x104d7e940 OCC::AbstractNetworkJob::start: !!! OCC::OcsShareJob created for  QUrl( "https://docker.solidgear.es:50013" )  querying ""
03-04 11:16:00:332 0x104d7e940 OCC::ShareDialog::slotSharesFetched: void OCC::ShareDialog::slotSharesFetched(const QString &) Status code:  100
03-04 11:16:02:372 0x104d7e940 OCC::AbstractNetworkJob::start: !!! OCC::CheckQuotaJob created for  QUrl( "https://docker.solidgear.es:50013" )  querying "/"
03-04 11:16:10:372 0x104d7e940 OCC::ConnectionValidator::checkAuthentication: # Check whether authenticated propfind works.
03-04 11:16:10:373 0x104d7e940 OCC::AbstractNetworkJob::start: !!! OCC::PropfindJob created for  QUrl( "https://docker.solidgear.es:50013" )  querying "/"

**Comment by lmaestro** [comment]
The same problem without encryption 
CC/ @dragotin

**Comment by lmaestro** [comment]
@dragotin I cant reproduce it using beta2, but do you any idea why this happened?

**Comment by lmaestro** [comment]
I can't share by link using password in OS X
![screen shot 2015-03-11 at 10 11 08 am](https://cloud.githubusercontent.com/assets/4689430/6593259/3a6251e6-c7d7-11e4-9d50-ff511d1b5c69.png)

**Comment by ckamm** [comment]
@lmaestro Works for me on Linux.

**Comment by dragotin** [comment]
@lmaestro: What exactly was the problem? The screenshot does not really indicate a problem.

**Comment by lmaestro** [comment]
@ckamm @dragotin It only happens on OS X when I try to share a file by link using password, it happens with the latest build 2115

**Comment by lmaestro** [comment]
It works fine in oc7.

**Comment by dragotin** [comment]
This is the server problem reported here: https://github.com/owncloud/core/issues/14506 

It's fixed, and will be in the new 8.0.2

**Comment by lmaestro** [comment]
Yes, it is fixed in  8.0.2.0. so I close this issue.
